Lib/asyncio/unix_events.py

```diff
@@ -51,8 +51,13 @@ def __init__(self, selector=None):
 
     def close(self):
         super().close()
-        for sig in list(self._signal_handlers):
-            self.remove_signal_handler(sig)
+        if not sys.is_finalizing():
+            for sig in list(self._signal_handlers):
+                self.remove_signal_handler(sig)
+        else:
+            warinigs.warn(f"Closing the loop {self} on interpreter shutdown "
```

Comment:
Please replace {self} with {self!r}. See for example BaseEventLoop.close() resource warning.